### PR TITLE
entropy: stm32: Move clock configuration to device tree

### PR DESCRIPTION
**Pull request by erwango**
Similarly to #53258, update entropy driver to get clock configuration from device tree.

Noticeable changes:
- STM32L4 are now using MSI or HSI48 instead of PLLSAI as RNG CLK
- A runtime check is implemented to check RNG clock config correctness

Based on #53258

### CODE REVIEW

**Review by desowin — DISMISSED** [review]

**Comment by erwango** [comment]
Rebased after https://github.com/zephyrproject-rtos/zephyr/pull/53258 was merged

File: dts/arm/st/f4/stm32f410.dtsi [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <zephyr/dt-bindings/clock/stm32f410_clock.h>
+#include <zephyr/dt-bindings/clock/stm32f4_clock.h>
```
Comment (gautierg-st) [inline dts/arm/st/f4/stm32f410.dtsi:7]: I don't think this change is right

**Review by FRASTM — DISMISSED** [review]

File: dts/arm/st/f4/stm32f446.dtsi [hunk 2]
```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/dt-bindings/clock/stm32f446_clock.h>
 #include <st/f4/stm32f401.dtsi>
 #include <zephyr/dt-bindings/clock/stm32f410_clock.h>
```
Comment (gautierg-st) [inline dts/arm/st/f4/stm32f446.dtsi:9]: I don't think including both bindings will work.

**Review by gautierg-st — CHANGES_REQUESTED** [review]

File: dts/arm/st/f4/stm32f410.dtsi (hunk 1 above)
Comment (erwango) [reply dts/arm/st/f4/stm32f410.dtsi:7]: Good catch. Removed the faulty commit

File: dts/arm/st/f4/stm32f446.dtsi (hunk 2 above)
Comment (erwango) [reply dts/arm/st/f4/stm32f446.dtsi:9]: Good catch. Removed the faulty commit

**Review by gautierg-st — APPROVED** [review]
LGTM

**Comment by erwango** [comment]
@desowin I had to rebase. Would you mind having a quick new look ?

**Review by FRASTM — APPROVED** [review]

**Comment by joelguittet** [comment]
@erwango hello

This PR currently breaks my project. I'm using rng on stm32l4 for mbedtls purpose.
How my dts should be modified following your PR ? Can you give an example ?

Note: I'm using nucleo_l4a6 board with the following overlay actually:

```
&rng {
	status = "okay";
};
```

Thanks for the clarification,
Joel

**Comment by joelguittet** [comment]
Auto-reply, hope this can help somebody else reading this PR, I have added:

```
&clk_hsi48 {
	/* HSI48 required for RNG */
	status = "okay";
};
```

This activate the HSI clock that is used for RNG peripheral.

**Pull request merged**